src/test/java/uk/gov/hmcts/sptribs/services/CaseManagementServiceTest.java

```diff
@@ -79,104 +78,102 @@ public void setUp() {
     }
 
     @Test
-    void testCicCreateCaseData() throws Exception {
-        String caseDataJson = loadJson(CASE_DATA_FILE_CIC);
-        CaseData caseData = mapper.readValue(caseDataJson, CaseData.class);
+    void shouldCreateCicCaseData() throws Exception {
+        final String caseDataJson = loadJson(CASE_DATA_FILE_CIC);
+        final CaseData caseData = mapper.readValue(caseDataJson, CaseData.class);
 
-        Map<String, Object> caseDataMap = new ConcurrentHashMap<>();
+        final Map<String, Object> caseDataMap = new ConcurrentHashMap<>();
         caseDataMap.put(CASE_DATA_CIC_ID, caseData);
 
-        AppsConfig.EventsConfig eventsConfig = new AppsConfig.EventsConfig();
+        final AppsConfig.EventsConfig eventsConfig = new AppsConfig.EventsConfig();
         eventsConfig.setCreateEvent("citizen-cic-create-dss-application");
 
         cicAppDetail.setEventIds(eventsConfig);
-        when(appsConfig.getApps()).thenReturn(Arrays.asList(cicAppDetail));
+        when(appsConfig.getApps()).thenReturn(List.of(cicAppDetail));
 
-        assertNotNull(cicAppDetail);
 
         when(authTokenGenerator.generate()).thenReturn(TEST_USER);
 
-        CaseDetails caseDetail = CaseDetails.builder().caseTypeId(CASE_DATA_CIC_ID)
+        final CaseDetails caseDetail = CaseDetails.builder().caseTypeId(CASE_DATA_CIC_ID)
             .id(TEST_CASE_ID)
             .jurisdiction(CommonConstants.ST_CIC_JURISDICTION)
             .data(caseDataMap)
             .build();
 
         when(caseApiService.createCase(CASE_TEST_AUTHORIZATION, caseData, cicAppDetail)).thenReturn(caseDetail);
 
-        CaseResponse caseResponse = CaseResponse.builder().caseData(caseDataMap).build();
-        DssCaseData dssCaseData =  mapper.readValue(caseDataJson, DssCaseData.class);
+        final CaseResponse caseResponse = CaseResponse.builder().caseData(caseDataMap).build();
+        final DssCaseData dssCaseData =  mapper.readValue(caseDataJson, DssCaseData.class);
         caseData.setDssCaseData(dssCaseData);
-        CaseResponse createCaseResponse = caseManagementService.createCase(CASE_TEST_AUTHORIZATION, caseData);
+        final CaseResponse createCaseResponse = caseManagementService.createCase(CASE_TEST_AUTHORIZATION, caseData);
+        final CaseData caseResponseData = (CaseData) createCaseResponse.getCaseData().get(CASE_DATA_CIC_ID);
+
+        assertNotNull(cicAppDetail);
+        assertNotNull(createCaseResponse);
```

Comment:
Two new test can be added to cover the exception 'Invalid Case type application. Please check the request.' for both create and update case.